azure/datalake/store/core.py

```diff
@@ -1229,9 +1232,8 @@ def flush(self, syncFlag='METADATA', force=False):
             logger.debug('Wrote %d bytes to %s' % (limit, self))
```

Comment:
Maybe do this as a part of initialization of stream with no data:

if self.first_write:
                _put_data_with_retry(
                    self.azure.azure,
                    'CREATE',
                    path=self.path.as_posix(),
                    data=data[:limit],
                    overwrite='true',
                    write='true',
                    syncFlag=syncFlagLocal,
                    leaseid=self.leaseid,
                    filesessionid=self.filesessionid)
                self.first_write = False